client/shared/src/main/scala/org/http4s/client/DefaultClient.scala

```diff
@@ -91,8 +92,8 @@ private[http4s] abstract class DefaultClient[F[_]](implicit F: MonadCancelThrow[
       req: Request[F]
   )(onError: Response[F] => F[Throwable])(implicit d: EntityDecoder[F, A]): F[A] = {
     val r = if (d.consumes.nonEmpty) {
-      val m = d.consumes.toList
-      req.addHeader(Accept(MediaRangeAndQValue(m.head), m.tail.map(MediaRangeAndQValue(_)): _*))
+      val m = d.consumes.toList.map(MediaRangeAndQValue(_))
+      req.addHeader(Accept(NonEmptyList.fromListUnsafe(m)))
```

Comment:
Actually, since we know it's safe, because we're in a branch off `nonEmpty` should I just construct like this?:
```scala
req.addHeader(Accept(NonEmptyList(m.head, m.tail)))
```

Reply:
Eh, I think it's fine. `fromListUnsafe` is very nearly this anyway.
https://github.com/typelevel/cats/blob/6ce36fbd543974a1d1ac66b6670c891efd22826c/core/src/main/scala/cats/data/NonEmptyList.scala#L751-L755